### ISSUE TITLE
`sam local generate-event s3` does not URL encode S3 Object Key

### ISSUE THREAD

**Issue by fffergal** [awslabs/aws-sam-cli#379]
For example:
```
sam local generate-event s3 --bucket whatever --key something/dt=2018-05-04/_SUCCESS
```
will not URL encode the equals sign to %3d, like a real SNS event does as per [the docs][eventdocs].

I imagine this could be a common case for AWS Glue output or adding partitions to AWS Athena tables.

[eventdocs]: https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html

**Comment by sanathkr** [comment]
Yes, this is a feature request I would say. You can also pass URL Encoded input if you want in the interim.

**Comment by fffergal** [comment]
Thank you for taking a look.

Considering SNS will never make an event which isn't URL encoded, I think it would be good normal behaviour for generate-event. If a feature request takes longer, could I suggest a faster docs/--help change to make it clear users should URL encode their own keys? I can imagine other people will make the same mistake I did without it.

I can look at making a PR for the feature or docs, if it's all in Python?

**Comment by jfuss** [comment]
With the update to `generate-event` the s3 key's will be [URL encoded](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/local/lib/generated_sample_events/event-mapping.json#L643).

Marking as waiting for release

**Comment by fffergal** [comment]
That's excellent, thank you! I'll look out for it.

**Comment by jfuss** [comment]
Closing this as it was released in [0.6.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.6.0)

**Comment by fffergal** [comment]
Great, thanks!
